arangod/Cache/PlainBucket.cpp

```diff
@@ -142,7 +142,10 @@ void PlainBucket::evict(CachedValue* value, bool optimizeForInsertion) {
 
 void PlainBucket::clear() {
   TRI_ASSERT(isLocked());
-  memset(this, 0, sizeof(PlainBucket));
+  _state.clear();
+  memset(this + sizeof(BucketState), 0,
+         sizeof(PlainBucket) - sizeof(BucketState));
+  _state.unlock();
```

Comment:
I don't see any problems with the assertion in `unlock` as there is the same `isLocked` assertion at the beginning of the method, and `_state.clear` retains the locked flag. However, this shows that the old code actually had a race condition since `memset` set `_state` to zero, thereby clearing the locked flag, and than continued to clear the rest of the now unlocked bucket.

Reply:
Good catch!